src/libcrun/linux.c

```diff
@@ -430,16 +430,85 @@ get_bind_mount (const char *src, libcrun_error_t *err)
   return get_and_reset (&open_tree_fd);
 }
 
+static pid_t
+create_userns_for_idmapped_mount (const char *options, libcrun_error_t *err)
+{
+  cleanup_free char *dup_options = xstrdup (options);
+  char *option, *saveptr = NULL;
+  cleanup_pid pid_t pid = -1;
+  pid_t xchg_pid;
+
+  pid = syscall_clone (CLONE_NEWUSER | SIGCHLD, NULL);
+  if (UNLIKELY (pid < 0))
+    return crun_make_error (err, errno, "clone");
+
+  if (pid == 0)
```

Comment:
this is a special process we use to get a reference to a newly created user namespace.

It is killed as soon as we take a reference for its user namespace and use for the mount.  The `prctl (PR_SET_PDEATHSIG, SIGKILL)` is to prevent the crun process is killed for any reason but this process will keep living

Reply:
ah i see got it now. thanks for the explanation.